src/node_main_instance.cc

```diff
@@ -30,13 +30,14 @@ NodeMainInstance::NodeMainInstance(Isolate* isolate,
   SetIsolateUpForNode(isolate_, IsolateSettingCategories::kMisc);
 }
 
-NodeMainInstance* NodeMainInstance::Create(
+std::unique_ptr<NodeMainInstance> NodeMainInstance::Create(
     Isolate* isolate,
     uv_loop_t* event_loop,
     MultiIsolatePlatform* platform,
     const std::vector<std::string>& args,
     const std::vector<std::string>& exec_args) {
-  return new NodeMainInstance(isolate, event_loop, platform, args, exec_args);
+  return std::unique_ptr<NodeMainInstance>(
```

Comment:
Completely optional suggestion: One could use `std::make_unique<NodeMainInstance>(...)` instead

Reply:
@addaleax I tried doing that. It doesn't compile because the static method NodeMainInstance::Create is trying to use the private constructor of NodeMainInstance.
Since NodeMainInstance::Create is a member method of NodeMainInstance (albeit static), it has visibility to the private constructor.
Whereas, if I use std::make_unique, it tries to forward the pointer to the private constructor and fails to compile (as the private constructor isn't visible outside the member methods of NodeMainInstance).